Oracle-Validation.md

```diff
@@ -0,0 +1,98 @@
+# Client-Side Oracle Validation
+
+# Table of Contents
+
+* [Abstract](#abstract)
+* [General Requirements](#general-requirements)
+* [Oracle-Discovery](#oracle-discovery)
+* [Oracle-Validation](#oracle-validation)
+
+
+## Abstract
+
+The DLC protocol relies on trust-minimized oracles mapping real-world events to signatures, those ones
+then confidentially re-mapped by participants to a range of contract outcomes. A participant expected
+execution of the DLC is thus function of the oracle well-behaving as announced before event
+realization. In this current protocol version, oracle behavior is assumed to be trusted. This level
+of trust is expressed by the client oracle selection policy, and should faithfully reflect user
+oracle preferences.
+
+Oracle signature usage requires confidence that the associated `oracle_public_key` is owned by the
+correct remote subject (person, system or organization). The ownership is asserted by having the
+client verifying a signature which is bound to an identity anchor (domain name, LN node pubkey, ...).
+The evaluation of identity anchor correctness is beyond the scope of this protocol. A DLC client may
+use this identity anchor as a descriptor for a reputation system.
+
+# Oracle Discovery
+
+Actually, no oracle discovery protocol support is mandated. The following requirements are generic
+requirements, which should be swayed from only if protocol has higher built-in privacy mechanisms.
+
+## Requirements
+
+A node:
+* MUST persist its keyring across restart/shutdown.
+* SHOULD fetch any `oracle_public_key` advertised by a "good reputation" oracle.
+
+## Rationale
+
+A DLC client should desynchronize any oracle pubkey fetching from DLC contract setting, otherwise
+the fetched entity may deduce an usage of oracle upcoming events. By fetching any `oracle_public_key`
+ available, a DLC client is hindering better potential oracle usage among a wider anonymous set. 
+"Good-reputation" is enforced by the client oracle selection policy and its definition is beyond this
+current specification.
+
+# `oracle_announcement` Authentication
+
+[`oracle_announcement`](https://github.com/discreetlogcontracts/dlcspecs/blob/master/Oracle.md#oracle-announcements)  
+authentication happens at `offer_dlc` reception, thus verifying binding between an event and an
```

Comment:
Isn't it also authenticated before `offer_dlc` construction?

Reply:
IIUC your question, your suggesting that a DLC client could pre-authenticated events as soon as they're publicly announced by an oracle but _before_ they're proposed as a contract by a counterparty ?

I think so, added a MAY likewise :
"MAY pre-validate `oracle_event` signature against the `oracle_public_key` at the time of its publication"

Beyond the really slight perf improvement, it might also avoid to fetch `oracle_public_key` on `offer_dlc` reception, that way preserve receiver DLC existence confidentiality toward the oracle.